wagtailmenus/models/menus.py

```diff
@@ -510,10 +510,15 @@ def prime_menu_items(self, menu_items):
                 """
                 This is a `MenuItem` for a custom URL. It can be classed as
                 'active' if the URL matches the request path.
+                If `apps_settings.CUSTOM_URL_SMART_ACTIVE_CLASSES == True`,
+                it may also be classed as 'ancestor' if the current URL is a
+                'child' of the custom link URL.
                 """
-                request_path = self.request.path
-                if apply_active_classes and item.link_url == request_path:
-                    setattr(item, 'active_class', app_settings.ACTIVE_CLASS)
+                if apply_active_classes:
+                    request = self.request
+                    active_class = item.get_active_class_for_request(request)
+                    if active_class:
+                        setattr(item, 'active_class', active_class)
```

Comment:
@inostia We can probably simplify this last bit quite a lot by just doing:

```
if apply_active_classes:
    active_class = item.get_active_class_for_request(request)
    setattr(item, 'active_class', active_class)
```
I can't imagine it being an issue if item.active_class is set to an empty string.

Reply:
👍  I wasn't sure if it would create a conflict or not to set it to an empty string.